### PR TITLE
Sort unmanaged & deleted resources in a predictable order

### PR DESCRIPTION
**Pull request by sundowndev**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #321
| ❓ Documentation  | no 

## Description

For now, resources were fetched in parallel which would cause the resource list order to be unpredictable. With these changes, the resources are always sorted by their Terraform Id.

### CODE REVIEW

**Review by lotoussa — APPROVED** [review]
lgtm

File: pkg/analyser/analyzer.go [hunk 1]
```diff
@@ -6,9 +6,10 @@ import (
 
 	resourceaws "github.com/cloudskiff/driftctl/pkg/resource/aws"
 
+	"github.com/r3labs/diff/v2"
```
Comment (moadibfr) [inline pkg/analyser/analyzer.go:9]: pretty sure you have something different to handle import sorting we need to check that together

File: pkg/analyser/analyzer_test.go [hunk 2]
```diff
@@ -940,23 +990,31 @@ func TestAnalyze(t *testing.T) {
...
 			}
 
-			unmanagedChanges, err := diff.Diff(result.Unmanaged(), c.expected.Unmanaged())
-			if err != nil {
-				t.Fatalf("Unable to compare %+v", err)
-			}
-			if len(unmanagedChanges) > 0 {
-				for _, change := range unmanagedChanges {
-					t.Errorf("%+v", change)
+			for i, expected := range c.expected.Unmanaged() {
```
Comment (moadibfr) [inline pkg/analyser/analyzer_test.go:993]: diff have an ooption to check slice ordering: 
>SliceOrdering ensures that the ordering of items in a slice is taken into account
https://github.com/r3labs/diff

File: pkg/resource/resource.go [hunk 3]
```diff
@@ -48,3 +49,8 @@ type NormalizedResource interface {
 func IsSameResource(rRs, lRs Resource) bool {
 	return rRs.TerraformType() == lRs.TerraformType() && rRs.TerraformId() == lRs.TerraformId()
 }
+
+func Sort(res []Resource) []Resource {
+	sort.SliceStable(res, func(i, j int) bool { return res[i].TerraformId() < res[j].TerraformId() })
```
Comment (moadibfr) [inline pkg/resource/resource.go:54]: might want to take type into account to have a more readable output

**Review by moadibfr — CHANGES_REQUESTED** [review]

File: pkg/analyser/analyzer_test.go (hunk 2 above)
Comment (sundowndev) [reply pkg/analyser/analyzer_test.go:993]: Yes indeed! thanks

**Review by moadibfr — APPROVED** [review]
LGTM

**Review by wbeuil — APPROVED** [review]
I'm fine with that

**Review by eliecharra — APPROVED** [review]

**Pull request merged**